### PR TITLE
[SYCL][XPTI] Add missing buffer constructor data

### PR DESCRIPTION
**Pull request by vladimirlaz**
Extend information about created buffer:
 - Host object used to create buffer;
 - A string representing the type of buffer element;
 - Buffer element size in bytes
 - Buffer dimensions number.
 - Buffer size for each dimension.

The test changes in intel/llvm-test-suite#788 should be merged after this PR is merged.

### CODE REVIEW

**Comment by vladimirlaz** [comment]
The failures are expected and the tests updated accordingly have passed: Jenkins/verify-test-suite-logs

**Review by alexbatashev — CHANGES_REQUESTED** [review]
We already expose this info via [edge_create](https://intel.github.io/llvm-docs/SYCLInstrumentationUsingXPTI.html#sycl-stream-sycl-notification-signatures) tracepoint. Why is this duplication needed?

File: sycl/source/detail/xpti_registry.cpp [hunk 1]
```diff
@@ -17,22 +17,15 @@ __SYCL_INLINE_NAMESPACE(cl) {
...
+xpti::trace_event_data_t *XPTIRegistry::createTraceEvent(
+    const void *Obj, const void *FuncPtr, uint64_t &IId,
+    const detail::code_location &CodeLoc, uint16_t TraceEventType) {
+  xpti::utils::StringHelper NG;
+  auto Name = NG.nameWithAddress<void *>(CodeLoc.functionName(),
+                                         const_cast<void *>(FuncPtr));
   xpti::payload_t Payload(
       Name.c_str(), (CodeLoc.fileName() ? CodeLoc.fileName() : ""),
-      CodeLoc.lineNumber(), CodeLoc.columnNumber(), (void *)Obj);
+      CodeLoc.lineNumber(), CodeLoc.columnNumber(), const_cast<void *>(Obj));
```
Comment (alexbatashev) [inline sycl/source/detail/xpti_registry.cpp:28]: Is it better to update this in XPTI directly, so that it stores a const pointer?

**Review by alexbatashev — COMMENTED** [review]
@vladimirlaz please provide the description and make sure the title of this PR matches its content

**Review by alexbatashev — APPROVED** [review]

**Pull request merged**